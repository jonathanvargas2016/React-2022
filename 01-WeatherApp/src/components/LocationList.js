import React from "react";
import WeatherLocation from "./WeatherLocation";
import PropTypes from 'prop-types';

const strToComponent = cities =>(
    //map --> transformaciones de arrays. **Es necesario utilizar una clave
    cities.map((city)=>(<WeatherLocation key={city} city={city}></WeatherLocation>))
)

const LocationList = ({cities}) => (
    <div>
        {strToComponent(cities)}
    </div>
)



LocationList.propTypes = {
    cities:PropTypes.array.isRequired
}


export default LocationList;