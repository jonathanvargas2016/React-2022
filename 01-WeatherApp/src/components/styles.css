
/*MEDIA QUERY*/
@media (max-width: 48em){
    .locationList{
        max-height: 30vh;
        overflow: auto;/*hace aparecer automaticamente la barra de scroll*/
        margin-bottom: 20px;
    }
}
