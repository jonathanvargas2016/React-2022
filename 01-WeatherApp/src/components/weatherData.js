import React from "react";
import WeatherTemperature from "./weatherTemperature";
import WeatherExtraInfo from "./weatherExtraInfo";

const WeatherData = () => (
    <div>
        <WeatherTemperature temperature={20} weatherState={'sun'} />
        <WeatherExtraInfo wind={'10m/s'} huminity={80} />
    </div>
)

export default WeatherData;
