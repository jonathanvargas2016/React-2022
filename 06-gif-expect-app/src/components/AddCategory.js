import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('')
    const handleInputChange = e => {
        setInputValue(e.target.value)
    }

    const handleSubmit = e => {
        e.preventDefault() // prevenir el comportamiento por defecto. refresh del navegador
        console.log('submit hecho')

        if (inputValue.trim().length > 2) {
            // setCategories(cats => [inputValue, ...cats ])
            onNewCategory(inputValue.trim())
            setInputValue('')
        }
    }
    return (

        <form onSubmit={handleSubmit}>
            <input type="text" placeholder='Buscar Gifts' value={inputValue} onChange={handleInputChange} />

        </form>

    )
}
AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
