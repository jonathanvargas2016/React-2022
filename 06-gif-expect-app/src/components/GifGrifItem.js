import React from 'react'

export const GifGrifItem = ({title, url}) => {
    return (
        <li className='card'>
            <img src={url} alt={title} />
            <p>{title} </p>
        </li>
    )
}
