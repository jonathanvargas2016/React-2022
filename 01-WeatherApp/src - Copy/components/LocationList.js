import React from "react";
import WeatherLocation from "./WeatherLocation";
import PropTypes from 'prop-types';
import './styles.css'

const LocationList = ({cities,onSelectedLocation}) => {
    const handleWeatherLocationClic = city => {
        console.log('handleWeatherLocationClic');
        onSelectedLocation(city);
    }

    const strToComponent = cities => {

        //map --> transformaciones de arrays. **Es necesario utilizar una clave
        return (cities.map(city => (
            <WeatherLocation key={city}
                             city={city}
                             onWeatherLocationClic={()=>{handleWeatherLocationClic(city)}}>
            </WeatherLocation>)
        ))
    }


    return (<div className='locationList'>
        {strToComponent(cities)}
    </div>)
}

LocationList.propTypes = {
    cities:PropTypes.array.isRequired,
    onSelectedLocation:PropTypes.func,
}

export default LocationList;