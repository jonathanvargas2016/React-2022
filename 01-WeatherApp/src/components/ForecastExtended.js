import React,{Component} from "react";
import PropTypes from 'prop-types';
import './styles.css';
import ForecastItem from './ForecastItem'
import TransformForecast from '../services/transformForecast'

// const days = [
//     'Lunes',
//     'Martes',
//     'Miercoles',
//     'Jueves',
//     'Viernes'
// ]
//
// const data = {
//     temperature:10,
//     weatherState:'normal',
//     humidity:20,
//     wind:'normal',
// }

class ForecastExtended extends Component {
    key = '927958a9f637a1a8b0e46e9118bf112b'
    url = 'http://api.openweathermap.org/data/2.5/forecast'

    constructor() {
        super();
        this.state = {
            forecastData:null,
        }
    }
    async componentDidMount() {
        const url_forecast = `${this.url}?q=${this.props.city}&appid=${this.key}`
        const res = await fetch(url_forecast)
        const data = await res.json()
        console.log(data)
        const forecastData = TransformForecast(data)
        this.setState({forecastData})
    }

    renderForecastItemDays = () =>{
        //array de dias
        // return days.map(day => (<ForecastItem weekDay={day}
        //                                       key={day}
        //                                       hour={10}
        //                                       data={data}>
        //
        //     </ForecastItem>
        // ))

        return (<h1>'Render Items'</h1>)

    }

    render = ()=> {

        const {city} = this.props;
        const {forecastData} = this.state;
        return(
            <div>
                <h2 className='forescast-title'>Pronostico extendido para {city}</h2>
                {forecastData?
                    this.renderForecastItemDays():this.renderProgress()}
            </div>
        )
    }
    renderProgress = ()=>{
        return (
            <h3>
                'Cargando Pronostico extendido...'
            </h3>
        )
    }
}

ForecastExtended.propType = {
    city:PropTypes.string.isRequired
}

export default ForecastExtended;
