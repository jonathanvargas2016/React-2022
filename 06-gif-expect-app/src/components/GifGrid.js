import React, { useEffect, useState } from 'react'
import { GifGrifItem } from './GifGrifItem'
import { getGifs } from '../Helpers/getGifs'



export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([])

    // getGifs() // *** PESIMA PRACTICA ***no colocar directamente una funcion en el functional component

    const getImages = async () => {
        const newImages = await getGifs(category)
        setImages(newImages)
    }


    useEffect(() => {
        getImages()
    }, []) // hook se dispara solo una vez cuando se construye este componente

    return (
        <>
            <h3>{category}</h3>
            <ol className='card-grid'>
                {
                    images.map((img) => {
                        return <GifGrifItem
                            key={img.id}
                            {...img} > 

                        </GifGrifItem>
                    })
                }
            </ol>
        </>
    )
}
