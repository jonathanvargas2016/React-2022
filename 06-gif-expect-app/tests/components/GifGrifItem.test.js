import { render, screen } from '@testing-library/react'
import { GifGrifItem } from '../../src/components/GifGrifItem'

describe('Pruebas en <GIfItem>', () => {

    const title = "Saitama"
    const url = "https://one-punch.com/saitama.jpg"

    test('debe de hacer match con el snapshot', () => {
        const { container } = render(<GifGrifItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })

    test('debe de mostrar la img con el url y el alt indicado', () => {

        render(<GifGrifItem title={title} url={url} />)

        // screen: ver el componente
        // screen.debug()

        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)

        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)

    })
    test('Debe de mostrar el titulo del documento', () => {

        render(<GifGrifItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy()

    })
})