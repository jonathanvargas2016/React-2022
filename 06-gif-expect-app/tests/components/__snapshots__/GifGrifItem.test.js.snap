// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GIfItem> debe de hacer match con el snapshot 1`] = `
<div>
  <li
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
       
    </p>
  </li>
</div>
`;
